netket/utils/array.py

```diff
@@ -48,3 +48,7 @@ def __array__(self, dtype: DType = None):
         if dtype is None:
             dtype = self.wrapped.dtype
         return self.wrapped.__array__(dtype)
+
+    @property
+    def shape(self) -> Shape:
+        return self.wrapped.shape
```

Comment:
I didn't want to add it because there was no immediate use case and it felt wrong to start adding some functions from `np.ndarray` but not others (why not `.transpose()` or `.ndim`, etc.).

OTOH, if there is a clear case for why this is much more comfortable in this PR, I'd be willing to reconsider.

Reply:
I think it ended up not to matter so we can walk back on this. However, `shape` is a much more fundamental property than most of the stuff numpy can calculate on an array, so I think it's justifiable to keep it. (It's a shame it can't have a catch-all interface that exposes the underlying array.)